### PR TITLE
Change extensions/v1beta to apps/v1 to be compliant with Kubernetes 1.16

### LINKED ISSUE #1237: Kubernetes: apiVersion extensions/v1beta1 is now apps/v1
As of Kuberentes v1.16 the api-version extensions/v1beta1 does not work anymore. 

It sould be updated to apps/v1.

### PR DESCRIPTION
**Pull request by WebSpider**
## What type of PR?

Enhancement

## What does this PR do?

Changes to Kubernetes YAML files to be able to use this in Kubernetes 1.16

### Related issue(s)
- Fixes #1237

## Prerequistes
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- N/A In case of feature or enhancement: documentation updated accordingly
- [X] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/guide.html#changelog) entry file.

### CODE REVIEW

**Comment by kaiyou** [comment]
This looks fine to me. @micw would have time to review as well?

**Comment by micw** [comment]
It will break compatibility with older kubernetes versions and not everyone has already updated. On the helm chart we can make it contitional to the k8s version or available APIs. For the static yamls the user must edit it accordingly.
Since the next release it in a few months (giving users time to upgrade their k8s), I'm fine with that breaking change.

**Comment by kaiyou** [comment]
@micw could you update your review for the merge to go?

**Comment by micw** [comment]
Already did, just forgot to approve

**Comment by garykrige** [comment]
Does this actually work for anyone? I get `error validating data: ValidationError(Deployment.spec): missing required field "selector" in io.k8s.api.apps.v1.DeploymentSpec;`

File: docs/kubernetes/mailu/ingress.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: extensions/v1beta1
+apiVersion: apps/v1
```
Comment (garykrige) [inline docs/kubernetes/mailu/ingress.yaml]: This does not work. It should be `networking.k8s.io/v1beta1`

**Comment by micw** [comment]
Oh, right. k8s is now also a bit picky about correct selectors ;)
Looks like we need one more PR.

File: docs/kubernetes/mailu/ingress.yaml (hunk 1 above)
Comment (micw) [reply docs/kubernetes/mailu/ingress.yaml]: Can you confirm that this works on 1.16? extensions/v1beta1 still works for me and it's going to finally change in 1.20 (https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/).

But apps/v1 is definitely wrong. I must have overlooked this when reviewing the change.

**Comment by micw** [comment]
You can find verified-working yamls at https://github.com/Mailu/helm-charts/tree/master/mailu/templates. Would you mind to create a new PR against master containing your findings?

File: docs/kubernetes/mailu/ingress.yaml (hunk 1 above)
Comment (garykrige) [reply docs/kubernetes/mailu/ingress.yaml]: yeah I've just upgraded to 1.16 and its working

**Comment by garykrige** [comment]
I'll make a PR at some point, but its frustrating for broken config to pass so many checks

**Comment by kaiyou** [comment]
k8s config still does not undego any automated test (and is not displayed as properly supported yet). If you find it frustrating, by all mean help us through the review process :)

**Pull request merged**